h/h_api/model/json_api.py

```diff
@@ -32,3 +33,69 @@ def create(
     @property
     def detail(self):
         return self.raw.get("detail", None)
+
+
+class JSONAPIData(Model):
+    """A single JSON API data object (call or response)."""
+
+    # TODO! - This would be nice but introduces a circular dependency with
+    # Schema as it needs the error stuff above via SchemaValdiationError
+    # schema = Schema.get_validator('json_api.json#/$defs/resourceObject')
+
+    @classmethod
+    def create(
+        cls, data_type, _id=None, attributes=None, meta=None, relationships=None
+    ):
+        return cls(
+            {
+                "data": cls.dict_from_populated(
+                    type=DataType(data_type).value,
+                    id=_id,
+                    attributes=attributes,
+                    meta=meta,
+                    relationships=relationships,
+                )
+            }
+        )
+
+    @property
+    def _data(self):
+        return self.raw["data"]
+
+    @property
+    def id(self):
+        return self._data["id"]
+
+    @property
+    def type(self):
+        """
+        The data type of this object
+        :rtype: DataType
+        """
+        return DataType(self._data["type"])
+
+    @property
+    def attributes(self):
+        """A dict of attributes for this object."""
+        return self._data["attributes"]
+
+    @attributes.setter
+    def attributes(self, attributes):
+        self._data["attributes"] = attributes
+
+    @property
+    def meta(self):
+        """The data metadata (not the root metadata)."""
+        return self._data.get("meta", {})
+
+    @property
+    def relationships(self):
+        """Relationships between this object and others"""
+        return self._data["relationships"]
+
+    @property
+    def id_reference(self):
+        """An id reference.
+
+        This is a custom extension to JSON API."""
+        return self.meta.get("$anchor")
```

Comment:
Not sure what this is

Reply:
I'll expand on this. This is a reference you can give to an object when sending it, when you don't know it's id yet, so you can refer to it in relationships. It's used for groups, because we use `groupid` as our main currency, which frustratingly aren't the id of the group.